### PR TITLE
Add more IMVC tests

### PR DESCRIPTION
**Pull request by Octogonapus**
Fix some weird edge-case bugs in the iterative controllers around getting error when disabled.

### Description of the Change

Tests :)
This also fixes some weird edge-case bugs in the Iterative controllers where `getError()` would not update properly if the controller was being stepped while disabled.

### Applicable Issues

None.

### CODE REVIEW

File: src/api/control/iterative/iterativeVelPidController.cpp [hunk 1]
```diff
@@ -103,7 +106,8 @@ double IterativeVelPIDController::getOutput() const {
 }
 
 double IterativeVelPIDController::getError() const {
-  return error;
+  return target - velMath->getVelocity().convert(rpm);
+  ;
```
Comment (Octogonapus) [inline src/api/control/iterative/iterativeVelPidController.cpp]: Extra semicolon 🤔
